### PR TITLE
Add sqliteodbc Driver for org.kicad.KiCad

### PR DESCRIPTION
**Pull request by jmaibaum**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project.
- [x] The KiCAD project owns the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build org.kicad.KiCad.ODBCDriver.sqliteodbc

File: org.kicad.KiCad.ODBCDriver.sqliteodbc.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+id: org.kicad.KiCad.ODBCDriver.sqliteodbc
```
Comment (hfiguiere) [inline org.kicad.KiCad.ODBCDriver.sqliteodbc.yml:1]: the extension point in beta is `org.kicad.KiCad.Nightly.ODBCDriver`

this is why this is failing.

**Comment by hfiguiere** [comment]
also you probably need to add a `flathub.json` with `skip-icon-checks: true`

Like this: https://github.com/flathub/org.gimp.GIMP.Plugin.GMic/blob/master/flathub.json

File: org.kicad.KiCad.ODBCDriver.sqliteodbc.yml (hunk 1 above)
Comment (jmaibaum) [reply org.kicad.KiCad.ODBCDriver.sqliteodbc.yml:1]: Ouch, thanks. I will prepare another PR over at the main repo then. The other Para can be left alone for now then... (#3819 and #3820)

**Comment by jmaibaum** [comment]
Thanks for the hint re `flathub.json`. I should have followed my previous extensions for KiCad...

File: org.kicad.KiCad.ODBCDriver.sqliteodbc.yml (hunk 1 above)
Comment (jmaibaum) [reply org.kicad.KiCad.ODBCDriver.sqliteodbc.yml:1]: PR https://github.com/flathub/org.kicad.KiCad/pull/134 has been merged and the beta build is done.

**Comment by jmaibaum** [comment]
This, and #3819 + #3820 should now build, as the extension mount point in the beta branch of org.kicad.KiCad has been fixed.

**Comment by hfiguiere** [comment]
you can trigger builds:

bot, build org.kicad.KiCad.ODBCDriver.sqliteodbc

**Comment by jmaibaum** [comment]
The build on the `beta` branch including the fix the extension point has finished successfull 16 hours ago, see: https://github.com/flathub/org.kicad.KiCad/blob/beta/org.kicad.KiCad.yml#L23, but it doesn't seem to be uploaded, since installing the flathub-beta version of org.kicad.KiCad, run a command line, and then check `/app/manifest.json`, there is still:

```json
  "add-extensions" : {
    "org.kicad.KiCad.Library" : {
      "directory" : "extensions/Library",
      "autodelete" : true,
      "merge-dirs" : "template;symbols;footprints;3dmodels",
      "no-autodownload" : true,
      "subdirectories" : true,
      "version" : "beta"
    },
    "org.kicad.KiCad.Nightly.ODBCDriver" : {
      "directory" : "extensions/ODBCDriver",
      "autodelete" : true,
      "add-ld-path" : "lib",
      "merge-dirs" : "lib;template",
      "no-autodownload" : true,
      "subdirectories" : true,
      "version" : "beta"
    }
  },
```

How often does the beta repo get updated?

**Comment by hfiguiere** [comment]
probably a downfall of other infra issues.

@barthalion any idea?

**Comment by hfiguiere** [comment]
Let's try if thing finally got solved

bot, build org.kicad.KiCad.ODBCDriver.sqliteodbc

File: org.kicad.KiCad.ODBCDriver.sqliteodbc.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        cat << EOF > ${FLATPAK_DEST}/template/sqliteodbc.ini
+        [SQLite3]
+        Description=SQLite3 ODBC Driver
+        Driver=/app/extensions/ODBCDriver/lib/libsqlite3odbc.so
+        Setup=/app/extensions/ODBCDriver/lib/libsqlite3odbc.so
+        Threading=2
+        EOF
+      - >
+        install -Dm644 --target-directory=${FLATPAK_DEST}/share/metainfo \
+          org.kicad.KiCadODBCDriver.sqliteodbc.metainfo.xml
```
Comment (hfiguiere) [inline org.kicad.KiCad.ODBCDriver.sqliteodbc.yml:32]:
```suggestion
          org.kicad.KiCad.ODBCDriver.sqliteodbc.metainfo.xml
```

**Comment by jmaibaum** [comment]
I have the beta branch of org.kicad.KiCad installed on my machine (still on the version without the fix). It did not receive an update via `flatpak update` yet.

**Comment by jmaibaum** [comment]
Thanks for fixing that stupid typo of mine @hfiguiere .

**Comment by jmaibaum** [comment]
bot, build org.kicad.KiCad.ODBCDriver.sqliteodbc

File: org.kicad.KiCad.ODBCDriver.sqliteodbc.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      - type: archive
+        strip-components: 2
+        url: http://www.ch-werner.de/sqliteodbc/sqliteodbc-0.9998.tar.gz
+        sha256: fabcbec73f98d1a34911636c02c29fc64147d27516b142e8e132c68c05a6065b
+        x-checker-data:
+          type: html
+          url: http://www.ch-werner.de/sqliteodbc/
+          version-pattern: <A HREF="sqliteodbc-([\d\.]+)\.tar\.gz">current source code</A>
+          url-template: http://www.ch-werner.de/sqliteodbc/sqliteodbc-$version.tar.gz
+      - type: file
```
Comment (hfiguiere) [inline org.kicad.KiCad.ODBCDriver.sqliteodbc.yml:48]: for the build file you need to add a source shell like this https://github.com/flathub/org.gimp.GIMP/blob/master/org.gimp.GIMP.json#L281

**Comment by jmaibaum** [comment]
bot, build org.kicad.KiCad.ODBCDriver.sqliteodbc

**Comment by jmaibaum** [comment]
Again, thanks for the help @hfiguiere

**Comment by jmaibaum** [comment]
Local tests using a test DB for KiCad are good. I consider this extension ready to be merged, now.

**Comment by jmaibaum** [comment]
bot, build org.kicad.KiCad.ODBCDriver.sqliteodbc

**Comment by jmaibaum** [comment]
@hfiguiere The KiCad project plans the stable release of 7.0.0 (where these extensions, along with #3819 and #3820 will be necessary for people to use the new "[Database libraries](https://docs.kicad.org/master/en/eeschema/eeschema.html#database-libraries)" feature), next week. Any chance to get these merged by then? Unitl then I wanted to test-drive them using the 7.0.0 RCs in beta, and then have them available in stable once the final release is out.

**Comment by jmaibaum** [comment]
All local tests went fine since all the builds succeeded, but testing is a little cumbersome reaching out to the test builds from the PRs (which get deleted after a couple of days)...

**Comment by jmaibaum** [comment]
To sum up again, I consider #3818, #3819, and #3820 ready to be merged for further testing in beta, until the stable release will come out next week.

**Comment by hfiguiere** [comment]
I have absolutely no power to merge anything. @barthalion @nedrichards can help with that.

**Pull request closed without merging**